components/match2/wikis/overwatch/match_group_input_custom.lua

```diff
@@ -0,0 +1,502 @@
+---
+-- @Liquipedia
+-- wiki=overwatch
+-- page=Module:MatchGroup/Input/Custom
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Json = require('Module:Json')
+local Logic = require('Module:Logic')
+local Lua = require('Module:Lua')
+local Opponent = require('Module:Opponent')
+local String = require('Module:StringUtils')
+local Table = require('Module:Table')
+local Variables = require('Module:Variables')
+
+local MatchGroupInput = Lua.import('Module:MatchGroup/Input', {requireDevIfEnabled = true})
+
+local _ALLOWED_STATUSES = { 'W', 'FF', 'DQ', 'L', 'D' }
+local _FINISHED_INDICATORS = { 'skip', 'np', 'cancelled', 'canceled' }
+local _MAX_NUM_OPPONENTS = 8
+local _MAX_NUM_PLAYERS = 10
+local _MAX_NUM_MAPS = 9
+local _DEFAULT_BESTOF = 3
+local _NO_SCORE = -99
+
+local _EPOCH_TIME = '1970-01-01 00:00:00'
+
+-- containers for process helper functions
+local matchFunctions = {}
+local mapFunctions = {}
+local opponentFunctions = {}
+
+local CustomMatchGroupInput = {}
+
+-- called from Module:MatchGroup
+function CustomMatchGroupInput.processMatch(_, match)
+	-- Count number of maps, and automatically count score
+	match = matchFunctions.getBestOf(match)
+	match = matchFunctions.getScoreFromMapWinners(match)
+
+	-- process match
+	Table.mergeInto(
+		match,
+		matchFunctions.readDate(match)
+	)
+	match = matchFunctions.getOpponents(match)
+	match = matchFunctions.getTournamentVars(match)
+	match = matchFunctions.getVodStuff(match)
+	match = matchFunctions.getExtraData(match)
+
+	return match
+end
+
+-- called from Module:Match/Subobjects
+function CustomMatchGroupInput.processMap(_, map)
+	map = mapFunctions.getExtraData(map)
+	map = mapFunctions.getScoresAndWinner(map)
+	map = mapFunctions.getTournamentVars(map)
+
+	return map
+end
+
+function CustomMatchGroupInput.processOpponent(record, date)
+	local opponent = Opponent.readOpponentArgs(record)
+		or Opponent.blank()
+
+	-- Convert byes to literals
+	if opponent.type == Opponent.team and opponent.template:lower() == 'bye' then
+		opponent = {type = Opponent.literal, name = 'BYE'}
+	end
```

Comment:
fyi this will only convert team opponent byes to literals
if the wiki should have non team opponents then it would not convert them to literals!

Reply:
Is there a major downside to this? What is the difference between a team that is converted to a literal and one that isn't?

Reply:
BYE's should always be literals
independent if team opponent or other (non literal) opponent
so if the wiki has other opponent types (than team and literal) for them the BYEs should probably be converted as well

Reply:
So would similar code need to be added to ``function CustomMatchGroupInput.processPlayer(_, player)`` (which currently just returns the player)?

Reply:
i would do the check on an opponent basis

imo kick
```lua
			if opponent.type == Opponent.team then
				opponent.icon = opponentFunctions.getIcon(opponent.template)
			end
```
in line 357-359 and then do the following at line 69 and following
```lua
	if opponent.type == Opponent.team then
		if opponent.template:lower() == 'bye' then
			opponent = {type = Opponent.literal, name = 'BYE'}
		else
			opponent.icon = opponentFunctions.getIcon(opponent.template)
		end
	elseif opponent.type ~= Opponent.literal then
		-- process other opponent types (especially in regards to their player inputs)
		-- ..........
		if Array.any(opponent.match2players, CustomMatchGroupInput._playerIsBye) then
			opponent = {type = Opponent.literal, name = 'BYE'}
		end
	end
..........
function CustomMatchGroupInput._playerIsBye(player)
	return (player.name or ''):lower() == 'bye' or (player.displayname or ''):lower() == 'bye'
end
```

(take it with a grain of salt i wrote it on mobile ...)

Reply:
Will test this out and see how it goes